### DISCUSSION TITLE
[feature]: Change manually transaction-id after channel opening

### DISCUSSION THREAD

**Discussion by M1ch43lV** [lightningnetwork/lnd#7149] [Q&A] [answered]
Problem: 
I bumped-fee a channel opening not with lncli but as noob with electrum wallet. 
LND does not recognize that the old transaction-id which has been replaced by the new one.

```
citadel@citadel:~ $ lncli pendingchannels  
{
    "total_limbo_balance": "0",
    "pending_open_channels": [
        {
            "channel": {
                "remote_node_pub": "03006fcf3312dae8d068ea297f58e2bd00ec1ffe214b793eda46966b6294a53ce6",
                "channel_point": "e0caa5ac212389c61aa9730b99570a45230164b3dfebab77c84508348de1bda4:0",
                "capacity": "20000",
                "local_balance": "18747",
                "remote_balance": "0",
                "local_chan_reserve_sat": "354",
                "remote_chan_reserve_sat": "354",
                "initiator": "INITIATOR_LOCAL",
                "commitment_type": "ANCHORS",
                "num_forwarding_packages": "0",
                "chan_status_flags": "",
                "private": false
            },
            "commit_fee": "593",
            "commit_weight": "772",
            "fee_per_kw": "528"
        }
    ],
    "pending_closing_channels": [
    ],
    "pending_force_closing_channels": [
    ],
    "waiting_close_channels": [
    ]
}
```
Link of the old transaction:
https://mempool.space/tx/e0caa5ac212389c61aa9730b99570a45230164b3dfebab77c84508348de1bda4

Link of the new transaction:
https://mempool.space/tx/e86648d149e85055af607f2c1ddbca4b5f0e86a7383a73e8e3db15d753616180

Solution:
Feature needed with replacement of channel_point to point to the new contract or even better LND updates automatically and replaces the old onchain transaction ID with the new one

Running LND v.0.15.4 on citadel environment
Bitcoin Knots 20220529 (Based on Bitcoin Core 23.0.0)

**Comment by alexbosworth** [comment, marked as answer]
The spec doesn't provide for negotiating replacement on open channels, recognizing the new outpoint wouldn't be sufficient to enable the new channel because the change in tx will invalidate the signatures

**Comment by saubyk** [comment]
hi @M1ch43lV moving this issue to a discussion, if you want to converse on this further to improve your understanding of what's possible with channel opening transactions. As is, this is not a feature request which can be supported, so closing the issue.

**Comment by M1ch43lV** [comment]
Ok, I realize the mistake I made. 

Taken from https://docs.lightning.engineering/the-lightning-network/payment-channels/lifecycle-of-a-payment-channel
"At its core, a payment channel is a 2-of-2 multisignature contract between two parties holding bitcoin. While this bitcoin is held cooperatively by the two parties from the perspective of the Blockchain, each party owns a portion of it, and maintains a record of this ownership locally."
I signed the transaction with the my own node's key with the help of electrum but the contract was not signed by the other node. Therefore it is interpreted as a withdrawal. The node holder is now lucky to receive 20k sats :-) In this case https://amboss.space :-)

So coming to my next question. How to remove the pending opening channel status from the LND database?  The possibly forever lasting pending status does not allow to open a new channel to the same node. Force close is not possible, as the contract has to be approved by both nodes before.

**Comment by alexbosworth** [comment]
can you clarify more details about what happened, it was a 20k sat channel?

**Reply by M1ch43lV** [reply]
Sure. Thank you for asking.

I opened the small 20k sat channel to amboss.space using thunderhub with 1vsat/byte. Because mempool was relatively full at the time of opening and because the channel opening was pending for longer than 3 hours I decided to use the RBF feature to accelerate the channel opening.
For security purposes I have exported the private key of my node to my electrum wallet in case of recovery of my onchain bitcoin funds. The tutorial I used is written in German: https://btc21.de/lightning-node-retten/

Since I am used to Electrum Wallet, I made use of the RBF feature there. I increased the fee to 2vsat/byte and signed the transaction with the node's key. This was the big mistake afterwards because the replacement of transaction was signed one-sided and it is now interpreted as a withdrawal not as a contract between the two nodes. See mempool transaction links mentioned before

Now I have a zombie pending opening channel status I would like to close or abandon.

**Comment by alexbosworth** [comment]
OK yeah there is no contract you are right

Before RBF you had in LND a pre-signed tx that paid from the unconfirmed tx back to your own wallet
After RBF you made another tx that paid to the same address but was a different id so the pre-signed tx can't be used

Opening a channel is paying to a 2:2 multisig with your key and the peer key, so without the pre-signed tx you can't unilaterally take the funds in that output back to your own control

If amboss were cooperative you could both use https://github.com/guggero/chantools to cooperatively spend the funds from the 2:2 out to your sole control, but since that is out of protocol for LN and LN spec currently contains nothing to cover this scenario, I wouldn't expect them to cooperate

**Reply by M1ch43lV** [reply]
I was lucky that I was not using RBF for a 5M channel with electrum 🥲
However any idea how I solve the problem with the stucked opening channel? Waiting 14 days or using ```lncli abandonchannel``` command in case that is working for this special context?

**Reply by alexbosworth** [reply]
yeah abandonchannel will delete a channel that will never confirm

**Reply by Jestopher-BTC** [reply]
Thanks for pointing us in the right direction! I can see the pending channel open and would like to cooperate to return these stuck funds. What steps would we need to take? Would we need to run abandonchannel as well or would that cause funds to simply not be shown as a pending channel open?

We're somewhat limited in our CLI operations since our node is being run on Voltage.

**Reply by alexbosworth** [reply]
guggero chantools can do it, it's not part of the specs though

**Reply by guggero** [reply]
https://github.com/guggero/chantools/blob/master/doc/zombierecovery.md
